Golang  Error : not enough arguments in call to proxmox.NewClient

Hi guys! 

I'm new with Golang and I'm trying to install Terraform Provider Proxmox on my systems. I'm following two tutorials but I have the same error for each. I tried on MacOS X and on Windows 10, but it's the same error, at the same step of the installation 

`axel@macbook-pro ~ % go get -v github.com/Telmate/terraform-provider-proxmox/cmd/terraform-provider-proxmox
github.com/Telmate/terraform-provider-proxmox/proxmox
# github.com/Telmate/terraform-provider-proxmox/proxmox
go/src/github.com/Telmate/terraform-provider-proxmox/proxmox/provider.go:108:30: not enough arguments in call to proxmox.NewClient
	have (string, nil, *tls.Config)
	want (string, *http.Client, *tls.Config, int)`

I'm following these tutorials : 

- https://notamax.be/proxmox-v6-cluster-k8s-kubeadm-via-terraform-cloud-init/
- https://norocketscience.at/provision-proxmox-virtual-machines-with-terraform/

I search this error on internet but I did'nt find anything useful. Can someone help please ? 

Thank u guys, have a nice day. Sorry for my bad english,

AlexFatta

--- a/docs/provider.md
+++ b/docs/provider.md
@@ -30,5 +30,6 @@ The following arguments are supported in the provider block:
 * `pm_otp` - (Optional; or use environment variable `PM_OTP`) The  2FA OTP code.
 * `pm_tls_insecure` - (Optional) Disable TLS verification while connecting.
 * `pm_parallel` - (Optional; defaults to 4) Allowed simultaneous Proxmox processes (e.g. creating resources).
+* `pm_timeout` - (Optional; defaults to 300) Timeout value (seconds) for proxmox API calls.
 
 Additionally, one can set the `PM_OTP_PROMPT` environment variable to prompt for OTP 2FA code (if required).

--- a/go.mod
+++ b/go.mod
@@ -3,6 +3,6 @@ module github.com/Telmate/terraform-provider-proxmox
 go 1.13
 
 require (
-	github.com/Telmate/proxmox-api-go v0.0.0-20200715182505-ec97c70ba887
+	github.com/Telmate/proxmox-api-go v0.0.0-20200729204648-0bb7d4b38186
 	github.com/hashicorp/terraform v0.12.10
 )

--- a/proxmox/provider.go
+++ b/proxmox/provider.go
@@ -69,6 +69,11 @@ func Provider() *schema.Provider {
 				Optional: true,
 				Default:  false,
 			},
+			"pm_timeout": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				Default:  300,
+			},
 			"pm_otp": &pmOTPprompt,
 		},
 
@@ -85,7 +90,7 @@ func Provider() *schema.Provider {
 }
 
 func providerConfigure(d *schema.ResourceData) (interface{}, error) {
-	client, err := getClient(d.Get("pm_api_url").(string), d.Get("pm_user").(string), d.Get("pm_password").(string), d.Get("pm_otp").(string), d.Get("pm_tls_insecure").(bool))
+	client, err := getClient(d.Get("pm_api_url").(string), d.Get("pm_user").(string), d.Get("pm_password").(string), d.Get("pm_otp").(string), d.Get("pm_tls_insecure").(bool), d.Get("pm_timeout").(int))
 	if err != nil {
 		return nil, err
 	}
@@ -100,12 +105,12 @@ func providerConfigure(d *schema.ResourceData) (interface{}, error) {
 	}, nil
 }
 
-func getClient(pm_api_url string, pm_user string, pm_password string, pm_otp string, pm_tls_insecure bool) (*pxapi.Client, error) {
+func getClient(pm_api_url string, pm_user string, pm_password string, pm_otp string, pm_tls_insecure bool, pm_timeout int) (*pxapi.Client, error) {
 	tlsconf := &tls.Config{InsecureSkipVerify: true}
 	if !pm_tls_insecure {
 		tlsconf = nil
 	}
-	client, _ := pxapi.NewClient(pm_api_url, nil, tlsconf)
+	client, _ := pxapi.NewClient(pm_api_url, nil, tlsconf, pm_timeout)
 	err := client.Login(pm_user, pm_password, pm_otp)
 	if err != nil {
 		return nil, err

--- a/proxmox/provisioner.go
+++ b/proxmox/provisioner.go
@@ -3,6 +3,7 @@ package proxmox
 import (
 	"context"
 	"fmt"
+	"strconv"
 	"time"
 
 	pxapi "github.com/Telmate/proxmox-api-go/proxmox"
@@ -31,6 +32,7 @@ func Provisioner() terraform.ResourceProvisioner {
 var currentClient *pxapi.Client
 
 func applyFn(ctx context.Context) error {
+	var pm_timeout int
 	data := ctx.Value(schema.ProvConfigDataKey).(*schema.ResourceData)
 	state := ctx.Value(schema.ProvRawStateKey).(*terraform.InstanceState)
 
@@ -45,7 +47,11 @@ func applyFn(ctx context.Context) error {
 	vmr.SetNode(targetNode)
 	client := currentClient
 	if client == nil {
-		client, err = getClient(connInfo["pm_api_url"], connInfo["pm_user"], connInfo["pm_password"], connInfo["pm_otp"], connInfo["pm_tls_insecure"] == "true")
+		pm_timeout, err = strconv.Atoi(connInfo["timeout"])
+		if err != nil {
+			pm_timeout = 300
+		}
+		client, err = getClient(connInfo["pm_api_url"], connInfo["pm_user"], connInfo["pm_password"], connInfo["pm_otp"], connInfo["pm_tls_insecure"] == "true", pm_timeout)
 		if err != nil {
 			return err
 		}

